### ISSUE TITLE
Question about self adversarial attack augmentation

### ISSUE THREAD

**Issue by JVD9kh96** [AlexeyAB/darknet#8130]
Hello, I was interested in the self adversarial attacks and came up with two questions:
1- As far as I understood from the Yolov4 paper, the self adversarial attack consists of two steps; at the first stage, the image data is fed to the network, and the gradient of the loss is calculated with respect to the input image. The image is updated due to the gradient, but the weights remain unchanged. In the second step, the updated image goes through the network, and this time the weights are updated. The question is, which formula is valid for updating the image?  "new_image = old_image + LR* grad" or "new_image = 
 old_image - LR * grad" (this is the same like updating the weights).

2- The learning rate scheduler is used to reduce the learning rate for better convergence during the training. Should the learning rate for adversarial attacks be increased or decreases during the training? 

Thanks

**Comment by thuangb** [comment]
In adversarial attack, you want to **maximize** the loss instead of **minimize** it, therefore the update rule (gradient ascent) for the image will be the opposite from normal update rule (gradient descent) for the weight so the formula should be `new_image = old_image + LR* grad`

**Comment by JVD9kh96** [comment]
> In adversarial attack, you want to **maximize** the loss instead of **minimize** it, therefore the update rule (gradient ascent) for the image will be the opposite from normal update rule (gradient descent) for the weight so the formula should be `new_image = old_image + LR* grad`

I'm thankful for your time and answer. Would you mind if I ask another question? 
Since I didn't know which update rule was valid, I tried both of them. I chose the adversarial learning rate to be the same as the weights update learning rate. The adversarial attack was applied randomly in 50 percent of the training steps. I also used a cosine annealing scheduler. I didn't use any other augmentation techniques to check for the effect of SAT, but there wasn't a significant difference between the mAP in the presence of SAT and without it (less than 1 percent). Am I doing something wrong, or does this technique not contribute much to the final mAP?

**Comment by thuangb** [comment]
adversarial training usually don't improve mAP (even worsen it sometimes). The only benefit of adversarial training is to make to model more robust and less overfit to the dataset

**Comment by thuangb** [comment]
You can refer to this [paper](https://arxiv.org/abs/2004.04320) and this [link](https://openai.com/blog/adversarial-example-research/) for more undertstanding

**Comment by JVD9kh96** [comment]
> You can refer to this [paper](https://arxiv.org/abs/2004.04320) and this [link](https://openai.com/blog/adversarial-example-research/) for more undertstanding

Thanks a lot. I really appreciate that.
